packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart

```diff
@@ -170,9 +170,43 @@ class CpuProfileData with Serializable {
 
   factory CpuProfileData.fromJson(Map<String, Object?> json_) {
     final json = _CpuProfileDataJson(json_);
+
+    // All CPU samples.
+    final samples = json.traceEvents ?? [];
+
+    // Sort the samples so we can compute the observed time difference between
+    // each sample.
+    samples.sort((a, b) => a.timestampMicros!.compareTo(b.timestampMicros!));
+
+    // To compute the median efficiently, we compute the median of groups of 5
+    // elements, and then grab the median of those medians by sorting, which
+    // brings us a linear time complexity.
+    final mediansOfGroupsOf5 = <int>[];
+    for (var i = 1; i + 5 < samples.length; i += 5) {
+      // The time diff between the sample at index and the previous sample.
+      int diff(int index) {
+        return samples[index].timestampMicros! -
+            samples[index - 1].timestampMicros!;
+      }
+
+      mediansOfGroupsOf5.add(
+        _median5(diff(i), diff(i + 1), diff(i + 2), diff(i + 3), diff(i + 4)),
+      );
+    }
+    mediansOfGroupsOf5.sort();
+
+    // Sort the remaining diffs and grab the median, or trust the given period
+    // if we don't have any data.
+    //
+    // See https://github.com/flutter/devtools/pull/8941 for more information.
+    final samplePeriod =
```

Comment:
can you add tests that test this logic to compute the median?

Reply:
Sure, will do this next